packages/libp2p/src/circuit-relay/transport/index.ts

```diff
@@ -285,7 +285,7 @@ class CircuitRelayTransport implements Transport {
       })
 
       log('new outbound transient connection %a', maConn.remoteAddr)
-      return await this.upgrader.upgradeOutbound(maConn, {
+      return this.upgrader.upgradeOutbound(maConn, {
```

Comment:
Was this flagged by the linter?  This sort of return function will introduce bugs:

```js
try {
  return await fn()
} catch (err) {
  // if fn() throws it will be caught here
}
```

vs:

```js
try {
  return fn()
} catch (err) {
  // if fn() throws it will NOT be caught here
}
```